docs/shared/upp.yaml

```diff
@@ -0,0 +1,42 @@
+upp:
+  execution:
+    batchargs:
+      export: NONE
+      nodes: 1
+      walltime: "00:05:00"
+    envcmds:
+      - module use /path/to/modulefiles
+      - module load runtime-module
+      - source /etc/profile.d/slurm.sh
+    executable: /path/to/upp.x
+    mpiargs:
+      - "--ntasks $SLURM_CPUS_ON_NODE"
+    mpicmd: srun
+  files_to_copy:
+    postxconfig-NT.txt: /path/to/postxconfig-NT.txt
+  files_to_link:
+    eta_micro_lookup.dat: /path/to/nam_micro_lookup.dat
+    params_grib2_tbl_new: /path/to/params_grib2_tbl_new
+  namelist:
+    base_file: /path/to/base.nml
+    update_values:
+      model_inputs:
+        datestr: "{{ validtime.strftime('%Y-%m-%d_%H:%M:%S') }}"
+        filename: /path/to/dynf{{ user.leadtime }}.nc
+        filenameflux: /path/to/phyf{{ user.leadtime }}.nc
```

Comment:
Actually, `leadtime` is computed in the `user:` block. The value injected by the driver is `validtime`, which is a Python `datetime` object, like `cycle`. My reasoning:

1. Injecting as few variables as possible is a benefit, as it keeps the implementation simpler.
2. Either `leadtime` or `validtime` could be derived from the other, given `cycle`. But I'm not sure `timedelta`, which I think would be needed for an `int` -> `datetime` conversion, will be available to Jinja2 in configs. Operations on two `datetime` objects are certainly available.
3. Of `leadtime` and `validtime`, the latter is more general: More values can be immediately derived from it.
4. The consistency of having two Python `datetime` objects, instead of one `datetime` and one `int`, may be helpful as a memory aid.
5. On the command line, conversely, `leadtime` is more ergonomic: Saying `--cycle 2024-05-08T18 --leadtime 167` is IMO much simpler than trying to figure out what the correct `--validtime` argument would be. So I settled on `leadtime` as a user input, and `validtime` as a Jinja2 variable.

Reply:
There are a couple of options to include here, but valid time is a derived piece of information from the command line input, and shouldn't be stored in the static config.yaml file. Here are a few thoughts: 

* We could support both `leadtime` and `validtime` as Jinja2 variables. If `leadtime` is a timedelta object already, I don't think Jinja has a problem with using it. I don't think Jinja would let you make a timedelta object without some additional context that includes the `timedelta` object.
* We could show this example using `{{ validtime - cycle }}` inline without the redirection to the `user:` section. That value won't get rendered until run time anyway, so it just means we will need to provide the user section to this 


I guess now that I'm walking through those points, it really just boils down to questioning what value the redirection adds.